hop/io.py

```diff
@@ -98,21 +98,20 @@ def open(self, url, mode="r", group_id=None, ignoretest=True, **kwargs):
         logger.debug("connecting to addresses=%s  username=%s  topics=%s",
                      broker_addresses, group_id, topics)
 
-        if topics is None:
-            raise ValueError("no topic(s) specified in kafka URL")
-
         if self.auth is not None:
             credential = select_matching_auth(self.auth, broker_addresses[0], username)
         else:
             credential = None
 
         if mode == "w":
-            if len(topics) != 1:
-                raise ValueError("must specify exactly one topic in write mode")
+            if topics is None or len(topics) != 1:
+                topics = [None]
```

Comment:
Is the idea here that if they set multiple topics, you set the topic to None in the Producer so they are forced to manually enter topic in subsequent write calls so there is no ambiguity?

Reply:
Exactly; the choice we made at the adc level was to store only one topic (https://github.com/astronomy-commons/adc-streaming/blob/b1a5a81c52d2752a9fd63b9717c121285646d09b/adc/producer.py#L89), as either there is a specific topic to which the user wants to always publish or they will have to specify it with ever `write`, and there didn't seem to be much point in artificially restricting the caller in that latter case to one of the predeclared topics, so we just forget what topics they specified except that there was not exactly one.